### ISSUE TITLE
🐛 Terraform 1.6.0 crashing

### ISSUE THREAD

**Issue by jacobwoffenden** [ministryofjustice/analytical-platform#1836]
### Describe the bug.

Terraform 1.6.0 crashes during validation causing workflows to fail

```
Run terraform validate -no-color
/home/runner/work/_temp/866eec23-03e0-4353-97fb-997b1903301e/terraform-bin validate -no-color

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Source: https://github.com/ministryofjustice/data-platform/actions/runs/6454061276/job/17559133154?pr=1801#step:11:18

### To Reproduce

1. Run a Terraform component with 1.6.0 binary

### Expected Behaviour

Terraform steps do not crash

### Additional context

We saw this behaviour while working on Modernisation Platform, the workaround was to pin at 1.5.7

@dms1981 raised this with Hashicorp (https://github.com/hashicorp/terraform/issues/33988 closed as dupe of https://github.com/hashicorp/terraform/issues/33977)

**Comment by jacobwoffenden** [comment]
>We've now merged some fixes that will be coming in the forthcoming v1.6.1 release.

As per https://github.com/hashicorp/terraform/issues/33977#issuecomment-1751060831

**Comment by jacobwoffenden** [comment]
https://github.com/hashicorp/terraform/releases/tag/v1.6.1 is now released which fixes this
